dsl_parser/framework/parser.py

```diff
@@ -148,32 +162,111 @@ def _traverse_schema(self, schema, parent_element):
                 iteritems():
             if k_holder.value not in parsed_names:
+                current_namespace = v_holder.namespace or namespace
                 self._traverse_element_cls(element_cls=elements.UnknownElement,
-                                           name=k_holder, value=v_holder,
-                                           parent_element=parent_element)
+                                           name=k_holder,
+                                           value=v_holder,
+                                           parent_element=parent_element,
+                                           namespace=current_namespace)
+
+    def _traverse_element_type_schema(self, schema, parent_element, namespace):
+
+        def set_namespace_get_input(namespace_value, input_name):
+            return utils.generate_namespaced_value(
+                namespace_value, input_name)
+
+        def set_namespace_get_attribute_property(namespace_value, func):
+            value = func[0]
+            if value not in functions.AVAILABLE_NODE_TARGETS:
+                func[0] = utils.generate_namespaced_value(
+                    namespace_value, value)
+            return func
+
+        def find_and_namespace_key(key, element, function_namespace):
+            """
+            This will traverse the element in search of the key,
+            and will run the set namespace function on the key's
+            values.
+            """
+            if hasattr(element, 'iteritems'):
+                for k, v in element.iteritems():
+                    if k == key:
+                        element[k] = function_namespace(namespace, v)
+                    if isinstance(v, dict):
+                        element[k] = \
+                            find_and_namespace_key(key,
+                                                   v,
+                                                   function_namespace)
+                    elif isinstance(v, list):
+                        for i in xrange(len(v)):
+                            element[k][i] = \
+                                find_and_namespace_key(
+                                    key,
+                                    v[i],
+                                    function_namespace)
+            return element
+
+        def set_leaf_namespace(leaf_namespace, element):
+            if leaf_namespace:
+                if isinstance(element._initial_value, str) and \
+                        element._initial_value not in \
+                        constants.USER_PRIMITIVE_TYPES and \
+                        element.add_namespace:
+                    leaf_namespace = element.namespace or leaf_namespace
+                    element._initial_value = utils.generate_namespaced_value(
+                            leaf_namespace, element._initial_value)
+                elif isinstance(element._initial_value, dict):
+                    element._initial_value = \
```

Comment:
Seems a little weird that we'd traverse the dict 3 times, instead of traversing it once and checking for all the intrinsic functions